bwapi/BWAPIClient/Source/Convenience.h

```diff
@@ -17,16 +17,29 @@ inline void StrTerminate(char (&str)[N])
 };
 
 template <size_t N>
-inline void StrCopy(char (&dst)[N], const char *src)
+inline void StrCopy(char (&dst)[N], const char *src, int length = -1)
 {
-  std::copy(src, src + N, dst);
-  StrTerminate(dst);
+	if (length == -1)
```

Comment:
~~~cpp
if (length == -1) length = std::strlen(src);
if (length > N) length = N;
~~~

However this is just re-implementing `std::strncpy`.